MapleServer2/PacketHandlers/Game/PartyHandler.cs

```diff
@@ -66,85 +78,120 @@ private void HandleInvite(GameSession session, PacketReader packet)
-                    leaderSession.Send(PartyPacket.UpdateHitpoints(session.Player));
+                //Send the party leader all the stuff for the joining player
+                leaderSession.Send(PartyPacket.Join(session.Player));
+                leaderSession.Send(PartyPacket.UpdatePlayer(session.Player));
+                leaderSession.Send(PartyPacket.UpdateHitpoints(session.Player));
 
-                    leaderSession.Send(PartyPacket.UpdateHitpoints(partyLeader));
+                leaderSession.Send(PartyPacket.UpdateHitpoints(partyLeader));
 
-                    partyLeader.PartyId = newParty.Id;
+                partyLeader.PartyId = newParty.Id;
 
-                    party = newParty;
-                }
+                party = newParty;
+            }
 
-                session.Player.PartyId = party.Id;
+            session.Player.PartyId = party.Id;
 
-                //Create existing party based on the list of party members
-                session.Send(PartyPacket.CreateExisting(partyLeader, party.Members));
-                session.Send(PartyPacket.UpdatePlayer(session.Player));
-                foreach (Player partyMember in party.Members)
+            //Create existing party based on the list of party members
+            session.Send(PartyPacket.CreateExisting(partyLeader, party.Members));
+            session.Send(PartyPacket.UpdatePlayer(session.Player));
+            foreach (Player partyMember in party.Members)
+            {
+                //Skip first character because of the scuffed Create packet. For now this is a workaround and functions the same.
+                if (partyMember.CharacterId != party.Members.First().CharacterId)
                 {
-                    //Skip first character because of the scuffed Create packet. For now this is a workaround and functions the same.
-                    if (partyMember.CharacterId != party.Members.First().CharacterId)
-                    {
-                        //Adds the party member to the UI
-                        session.Send(PartyPacket.Join(partyMember));
-                    }
-                    //Update the HP for each party member.
-                    session.Send(PartyPacket.UpdateHitpoints(partyMember));
+                    //Adds the party member to the UI
+                    session.Send(PartyPacket.Join(partyMember));
                 }
-                //Sometimes the party leader doesn't get set correctly. Not sure how to fix.
+                //Update the HP for each party member.
+                session.Send(PartyPacket.UpdateHitpoints(partyMember));
             }
-            else
+        }
+
+        private void HandleLeave(GameSession session, PacketReader packet)
+        {
+            Party party = GameServer.PartyManager.GetPartyById(session.Player.PartyId);
+            session.Send(PartyPacket.Leave(session.Player, 1)); //1 = You're the player leaving
```

Comment:
Maybe move this under the check below so we don't even send the leave packet to users that don't even have a valid party

Reply:
I put it like this because it's possible to become "stuck" in a party when testing, this prevents you from having either state (ie. In party client-side but not for the server, or vice versa)